consumer/consumererror/README.md

```diff
@@ -0,0 +1,257 @@
+# Consumer errors
+
+This package contains error types that should be returned by a consumer when an
+error occurs while processing telemetry. The error types included in this
+package provide functionality for communicating details about
+the error for use upstream in the pipeline. Ideally the error returned by a
+component in its `consume` function should be from this package.
+
+## Error classes
+
+**Retryable**: Errors are retryable if re-submitting data to a sink may result
+in a successful submission.
+
+**Permanent**: Errors are permanent if submission will always fail for the
+current data. Errors are considered permanent unless they are explicitly marked
+as retryable.
```

Comment:
For permanent errors, we should probably always return the "number of not accepted" entries? Is that reasonable?

Reply:
I don't think we need to do that since the caller will already know how many items failed.